lib/connectors_async/job.rb

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'connectors_shared/job_status'
+
+# The class is actually supported for single-threaded usage EXCEPT for :documents field
+# :documents are a Queue that's stated to be safe in a threaded environment
+module ConnectorsAsync
+  class Job
+    class StatusUpdateError < StandardError; end
+    def initialize(job_id)
+      @data = {
+        :job_id => job_id,
+        :status => ConnectorsShared::JobStatus::CREATED,
+        :documents => Queue.new
+      }
+    end
```

Comment:
Instead of using a hash `@data` instance variable, why not use variables like `job_id`, `status`, `doc_queue`, etc...?

Reply:
it was done with assumption that sometime in future the job has to be stored persistently somewhere, therefore `@data` hash represents future model of data for jobs.

Reply:
For that, we can have a `save` method in the future to persist what we think should be persisted in the future, if needed (Not necessarily all fields in `@data` need to be persisted)

Reply:
Yeah indeed would be nice to have it in future and I expect it will change the model of jobs and everything around it significantly, so I'm not putting too much thought here for now :)